### PR TITLE
feat: Update NSTemplateSet CRD

### PR DESCRIPTION
**Pull request by mfrancisc**
## Description
This PR adds new fields to the `NSTemplateSet.Status` , those will be used to calculate the diff between the old TierTemplateRevision CR and the new one. 

For more details see: [Design doc](https://docs.google.com/document/d/1x5SoBT80df9fmVsaDgAE6DE7hE6lzmNIK087JUmgaJs/edit#heading=h.4gcyagfkx0m5)

Jira: https://issues.redhat.com/browse/KUBESAW-137 

## Checks
1. Did you run `make generate` target? **yes**

2. Did `make generate` change anything in other projects (member-operator)? **yes**

3. In case of **new** CRD, did you the following? **N/A**

4. In case other projects are changed, please provides PR links.
    - member-operator: https://github.com/codeready-toolchain/member-operator/pull/596

### CODE REVIEW

File: api/v1alpha1/nstemplateset_types.go [hunk 1]
```diff
@@ -86,6 +86,28 @@ type NSTemplateSetSpaceRole struct {
 // NSTemplateSetStatus defines the observed state of NSTemplateSet
 // +k8s:openapi-gen=true
 type NSTemplateSetStatus struct {
+	// The namespace templates provisioned for the current NSTemplateSet CR
+	// +optional
+	// +listType=atomic
+	Namespaces []NSTemplateSetNamespace `json:"namespaces"`
```
Comment (alexeykazakov) [inline api/v1alpha1/nstemplateset_types.go:92]:
```suggestion
	Namespaces []NSTemplateSetNamespace `json:"namespaces,omitempty"`
```

File: api/v1alpha1/nstemplateset_types.go [hunk 2]
```diff
@@ -86,6 +86,28 @@ type NSTemplateSetSpaceRole struct {
 // NSTemplateSetStatus defines the observed state of NSTemplateSet
 // +k8s:openapi-gen=true
 type NSTemplateSetStatus struct {
+	// The namespace templates provisioned for the current NSTemplateSet CR
```
Comment (MatousJobanek) [inline api/v1alpha1/nstemplateset_types.go:89]:
```suggestion
	// The namespace templates that were used last time to provision NSTemplateSet CR
```

File: api/v1alpha1/nstemplateset_types.go [hunk 3]
```diff
@@ -86,6 +86,28 @@ type NSTemplateSetSpaceRole struct {
...
+	// +optional
+	// +listType=atomic
+	Namespaces []NSTemplateSetNamespace `json:"namespaces"`
+
+	// the cluster resources template (for cluster-wide quotas, etc.) provisioned for the current NSTemplateSet CR
```
Comment (MatousJobanek) [inline api/v1alpha1/nstemplateset_types.go:94]:
```suggestion
	// The cluster resources template (for cluster-wide quotas, etc.) that was used last time to provision the NSTemplateSet CR
```

File: api/v1alpha1/nstemplateset_types.go [hunk 4]
```diff
@@ -86,6 +86,28 @@ type NSTemplateSetSpaceRole struct {
...
+	// +optional
+	ClusterResources *NSTemplateSetClusterResources `json:"clusterResources,omitempty"`
+
+	// the role template and the users to whom the templates was applied for the current NSTemplateSet CR
```
Comment (MatousJobanek) [inline api/v1alpha1/nstemplateset_types.go:98]:
```suggestion
	// The SpaceRole template and the users to whom the template was applied for when the NSTemplateSet CR was provisioned for the last time 
```

File: api/v1alpha1/nstemplateset_types.go [hunk 5]
```diff
@@ -86,6 +86,28 @@ type NSTemplateSetSpaceRole struct {
...
+	// +optional
+	// +listType=atomic
+	SpaceRoles []NSTemplateSetSpaceRole `json:"spaceRoles,omitempty"`
+
+	// FeatureToggles holds the list of feature toggles/flags enabled for the current NSTemplateSet CR
```
Comment (MatousJobanek) [inline api/v1alpha1/nstemplateset_types.go:103]:
```suggestion
	// FeatureToggles holds the list of feature toggles/flags that were enabled when the NSTemplateSet CR was provisioned for the last time
```

File: api/v1alpha1/nstemplateset_types.go [hunk 6]
```diff
@@ -86,6 +86,28 @@ type NSTemplateSetSpaceRole struct {
...
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=name
+	FeatureToggles []FeatureToggle `json:"featureToggles,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
```
Comment (MatousJobanek) [inline api/v1alpha1/nstemplateset_types.go:109]: the `FeatureToggle` type is used to configure the toggle, so it also contains the `Weight` field - we don't want to include that into the NSTemplateSet. 
I believe that a simple array of strings should be sufficient.

**Review by MatousJobanek — COMMENTED** [review]
Looks good, but let's not reuse the `FeatureToggle` type

File: api/v1alpha1/nstemplateset_types.go (hunk 1 above)
Comment (mfrancisc) [reply api/v1alpha1/nstemplateset_types.go:92]: thanks

File: api/v1alpha1/nstemplateset_types.go (hunk 6 above)
Comment (mfrancisc) [reply api/v1alpha1/nstemplateset_types.go]: Good point! Thanks for the suggestions.

Done in https://github.com/codeready-toolchain/api/pull/445/commits/1507f404e917c785b00a6010390560f565ba12a1

**Review by alexeykazakov — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**